cabal.config

```diff
@@ -1,32 +1,43 @@
 constraints: array ==0.5.1.0,
              base ==4.8.2.0,
              base16-bytestring ==0.1.1.6,
+             basement ==0.0.7,
              binary ==0.7.5.0,
              byteable ==0.1.1,
              bytestring ==0.10.6.0,
              containers ==0.5.6.2,
-             cryptohash ==0.11.7,
+             cryptohash ==0.11.9,
+             cryptonite ==0.25,
              deepseq ==1.4.1.1,
              directory ==1.2.2.0,
-             exceptions ==0.8.2.1,
+             exceptions ==0.10.0,
              filepath ==1.4.0.0,
+             foundation ==0.0.20,
+             generic-deriving ==1.12.2,
              ghc-prim ==0.4.0.0,
+             hashable ==1.2.7.0,
              integer-gmp ==1.0.0.0,
+             memory ==0.14.16,
              mmap ==0.5.9,
-             mtl ==2.2.1,
+             mtl ==2.2.2,
              pretty ==1.1.2.0,
-             primitive ==0.6.1.0,
+             primitive ==0.6.4.0,
+             random ==1.1,
              reinterpret-cast ==0.1.0,
              rts ==1.0,
-             stm ==2.4.4.1,
-             tar ==0.5.0.1,
+             semigroups ==0.18.5,
+             stm ==2.5.0.0,
+             tagged ==0.8.6,
+             tar ==0.5.1.0,
              template-haskell ==2.10.0.0,
-             temporary ==1.2.0.4,
-             text ==1.2.2.1,
+             temporary ==1.3,
+             text ==1.2.3.1,
+             th-abstraction ==0.2.8.0,
              time ==1.5.0.1,
              transformers ==0.4.2.0,
-             transformers-compat ==0.5.1.4,
+             transformers-compat ==0.6.2,
              unix ==2.7.1.0,
-             vector ==0.11.0.0,
-             vector-algorithms ==0.7.0.1,
-             zlib ==0.6.1.1
+             unordered-containers ==0.2.9.0,
+             vector ==0.12.0.2,
+             vector-algorithms ==0.8.0.1,
+             zlib ==0.6.2
```

Comment:
So many new transitive dependencies!

This is actually a good reason to commit this file for libraries. Not just to allow the dev environment to be replicated, but to have a record of the transitive dependencies we observed during development. Otherwise we might not even notice how big the list is growing.